master/internal/experiment.go

```diff
@@ -373,7 +376,89 @@ func (e *experiment) Receive(ctx *actor.Context) error {
 		}
 
 		ctx.Log().Info("experiment shut down successfully")
+
+		// for each operation we have 3 sets of experiment states:
+		// 1. experiment in desired target states
+		// 2. experiment in applicable states
+		// 3. the rest
+	case *apiv1.PauseExperimentRequest:
+		var err error
+		switch {
+		case e.State == model.PausedState:
+			ctx.Respond(&apiv1.PauseExperimentResponse{})
+		case model.RunningStates[e.State]:
+			if _, err = e.updateState(ctx, model.PausedState); err == nil {
+				ctx.Respond(&apiv1.PauseExperimentResponse{})
+			}
+		default:
+			err = status.Errorf(codes.InvalidArgument, "experiment in incompatible state %s", e.State)
+		}
+		if err != nil {
+			ctx.Respond(err)
+			ctx.Log().Error(err)
+		}
+
+	case *apiv1.ActivateExperimentRequest:
+		var err error
+		switch e.State {
+		case model.ActiveState:
+			ctx.Respond(&apiv1.ActivateExperimentResponse{})
+		case model.PausedState:
+			if _, err = e.updateState(ctx, model.ActiveState); err == nil {
+				ctx.Respond(&apiv1.ActivateExperimentResponse{})
+			}
+		default:
+			err = status.Errorf(codes.InvalidArgument, "experiment in incompatible state %s", e.State)
+		}
+		if err != nil {
+			ctx.Respond(err)
+			ctx.Log().Error(err)
+		}
+
+	case *apiv1.CancelExperimentRequest:
+		var err error
+		switch {
+		case model.StoppingStates[e.State] || model.TerminalStates[e.State]:
+			ctx.Respond(&apiv1.ActivateExperimentResponse{})
+		case model.RunningStates[e.State]:
+			if _, err = e.updateState(ctx, model.StoppingCanceledState); err == nil {
+				ctx.Respond(&apiv1.ActivateExperimentResponse{})
+			}
+		default:
+			err = status.Errorf(codes.InvalidArgument, "experiment in incompatible state %s", e.State)
+		}
+		if err != nil {
+			ctx.Respond(err)
+			ctx.Log().Error(err)
+		}
+		for _, child := range ctx.Children() {
+			ctx.Tell(child, killTrial{})
+		}
+
+	case *apiv1.KillExperimentRequest:
+		var err error
+		switch {
+		case model.StoppingStates[e.State] || model.TerminalStates[e.State]:
+			ctx.Respond(&apiv1.ActivateExperimentResponse{})
+		case model.RunningStates[e.State]:
+			if _, err = e.updateState(ctx, model.StoppingCanceledState); err == nil {
+				ctx.Respond(&apiv1.ActivateExperimentResponse{})
+			}
+		default:
+			err = status.Errorf(codes.InvalidArgument, "experiment in incompatible state %s", e.State)
+		}
+		if err != nil {
+			ctx.Respond(err)
+			ctx.Log().Error(err)
+		}
+		for _, child := range ctx.Children() {
+			ctx.Tell(child, killTrial{})
+		}
+
+	default:
```

Comment:
blocking: This will change the experiment actor semantics potentially. We should remove this case statement.